woocommerce/woocommerce-ios
# Shipping Labels: Package type selection and default values for custom package creation UI

Part of: #4743

## Description

In the shipping label creation flow, this adds to the UI (started in #4799) to enter the details for a new custom package to use for your shipping label. In this PR:

* The "Package Type" field is now active, so you can select the package type from a list.
* The default values for the form are set:
   * The length and weight units are set to the units in the store's settings.
   * The package type is set to "Box" by default.
   * Other fields (package dimensions and weight) are empty.

## Changes

* A new `PackageType` enum is added as an extension to `ShippingLabelCustomPackageFormViewModel` to represent the available package types. (In the networking model `ShippingLabelCustomPackage` this is represented simply by the `bool` property `isLetter`.)
* A `SelectionList` is added to `ShippingLabelCustomPackageForm` to select the package type.
* The expected default values are set in the `ShippingLabelCustomPackageFormViewModel` initializer.
* The view models for `ShippingLabelAddNewPackage` and `ShippingLabelCustomPackageForm` are now designated as `StateObject` to prevent them from losing state when the view is redrawn (e.g. after closing the selection list).
* A basic unit test is added for `ShippingLabelCustomPackageFormViewModel`, along with a `MockShippingSettingsService` to facilitate testing.

Before|After|After (package type)
-|-|-
![Add New Package - Custom Package - Before](https://user-images.githubusercontent.com/8658164/129262044-4b8741f8-72e1-4741-8be2-c6a35fdde48d.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-08-18 at 00 10 16](https://user-images.githubusercontent.com/8658164/129812312-abdcee93-9425-49ab-a6d8-610389da1aa9.png)|![Simulator Screen Shot - iPhone 11 Pro - 2021-08-18 at 00 10 19](https://user-images.githubusercontent.com/8658164/129812316-a85eb4c4-4506-430b-960e-295c878dfcbb.png)


## Testing

1. Make sure you have configured the WooCommerce WCShip plugin, and you have one or more packages set up in WP Admin under WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Launch the app in debug mode (so the feature flag is enabled).
3. Open an order detail that is eligible for the shipping label.
4. Tap the button "Create Shipping Label.”
5. Confirm "Ship From" and "Ship To.”
6. In the package details section, tap on “Package Selected.”
7. On the Package Selected screen, tap the “Create new package” button at the bottom of the screen.
8. On the "Add New Package" screen, confirm that "Package Type" defaults to "Box" and you can tap that field to select the package type.
9. Confirm that your store's length unit (e.g. "in" or "cm") appears next to the number field in the "Length," "Width," and "Height" rows.
10. Confirm that your store's weight unit (e.g. "oz" or "kg") appears next to the number field in the "Empty Package Weight" row.
11. Confirm all of the fields aside from package type default to empty and the expected placeholders appear.

## Submitter Checklist

Update release notes:
